lib/updatehub/shell.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018 O.S.Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_UPDATEHUB_SHELL)
+
+#include <shell/shell.h>
+#include <flash.h>
+#include <dfu/mcuboot.h>
+#include <dfu/flash_img.h>
+#include <updatehub.h>
+#include "firmware.h"
+#include "device.h"
+
+
+#if defined(CONFIG_UPDATEHUB_CE_SERVER)
+#define UPDATEHUB_SERVER CONFIG_UPDATEHUB_SERVER
+#else
+#define UPDATEHUB_SERVER "coap.updatehub.io"
+#endif
+
+static int cmd_run(const struct shell *shell, size_t argc,
+		   char **argv)
+{
+	int ret = -1;
+
+	shell_fprintf(shell, SHELL_INFO, "Starting UpdateHub run...\n");
+
+	switch (updatehub_probe()) {
+	case UPDATEHUB_HAS_UPDATE:
+		switch (updatehub_update()) {
+		case UPDATEHUB_OK:
+			ret = 0;
+			break;
+		default:
+			shell_fprintf(shell, SHELL_ERROR, "Error installing update.\n");
+			break;
+		}
+		break;
+
+	case UPDATEHUB_NO_UPDATE:
+		shell_fprintf(shell, SHELL_INFO, "No update found\n");
+		ret = 0;
+		break;
+
+	default:
+		shell_fprintf(shell, SHELL_ERROR, "Invalid response\n");
+		break;
+	}
+
+	return ret;
+}
+
+static int cmd_info(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	char *device_id = k_malloc(DEVICE_ID_SIZE);
+	char *firmware_version = k_malloc(BOOT_IMG_VER_STRLEN_MAX);
+
+	updatehub_get_firmware_version(firmware_version);
+	updatehub_get_device_identity(device_id);
```

Comment:
Can add 2nd buffer length param here and pass in `DEVICE_ID_SIZE`.

Reply:
If your comment about ``lib/updatehub/device.h`` is in a "resolved" section, could I guess assume this comment was resolved too?

Reply:
Sorry, I understood now!